src/main/java/com/github/martynagil/storemanagement/UserInterface.java

```diff
@@ -11,7 +17,65 @@ public UserInterface(Console console, ProductService productService) {
         this.productService = productService;
     }
 
-    public void exit() {
+    public void mShowProducts() {
+        showProducts(productService.getAllProducts());
+    }
+
+    public void mAddProduct() {
+        String name = console.askForString("Enter the name of product: ");
+        String brand = console.askForString("Enter the name of brand: ");
+        String type = console.askForString("Enter the type: ");
+        String barcode = console.askForString("Enter the barcode: ");
+        double price = console.askForDouble("Enter the price: ");
+        Product product = new Product(name, brand, type, barcode, price);
```

Comment:
a dopuszczasz do utworzenia produktu który ma pustą nazwę np.? Tak walidacja też byłaby przydatna

Reply:
nie dopuszczam.

Reply:
no to na tym poziomie (`UserInterface`) musiałabyś sprawdzać czy użytkownik podaje odpowiednią wartość. Bo z pkt widzenia konsoli jest string i jest w porządku, ale biznesowo to nie są dobre dane